tracing-attributes/src/lib.rs

```diff
@@ -364,18 +355,30 @@ fn gen_body(
             })
             .map(|(user_name, real_name)| quote!(#user_name = tracing::field::debug(&#real_name)))
             .collect();
+
+        // when async-trait is in use, replace instances of "self" with "_self" inside the fields values
+        if let Some(async_trait_fun) = async_trait_fun {
+            if let Some(Fields(ref mut fields)) = args.fields {
+                for mut field in fields.iter_mut() {
+                    if let Some(ref mut e) = field.value {
+                        syn::visit_mut::visit_expr_mut(
+                            &mut SelfReplacer {
+                                ty: async_trait_fun.self_type.clone(),
+                            },
+                            e,
+                        );
+                    }
+                }
+            }
+        }
```

Comment:
Take it or leave it: I think we can express this with a little less rightward drift, like this:
```suggestion
        if let (Some(async_trait_fun), Some(Fields(ref mut fields))) = (async_trait_fun, args.fields) {
            let mut replacer = SelfReplacer {
                ty: async_trait_fun.self_type.clone(),
            };
            for e in fields.iter_mut().filter_map(|f| f.value.as_mut()) {
                syn::visit_mut::visit_expr_mut(&mut replacer, e);
            }
        }
```